### PR TITLE
[Fix] Record real cost for Roomote inference usage

### PR DESCRIPTION
**Pull request by mrubens**
Trial tasks tracked tokens but recorded $0 cost everywhere in the app while OpenRouter billed normally (observed on the first staging trial tenant: 77 usage events, all `cost_micro_usd = 0`, key burndown at $0.22).

## Why

OpenCode prices messages from its own models.dev catalog, which cannot see the custom `roomote` provider — and the seeded trial models carried no pricing metadata for the app to supply instead. Result: `opencode_message` cost of 0 on every trial event, so task cards and cost analytics show nothing. (The $5 cap and the Settings credits burndown were unaffected — they read the key's own `limit_remaining`.)

## How

- **models.dev lookups resolve `roomote/...` ids through their OpenRouter upstream** (`rebaseRoomoteModelIdToUpstream`), so metadata refresh and seeding find pricing and context windows for trial models.
- **Choosing the trial seeds its models with catalog metadata up front** (best-effort with a 10s timeout; a failed fetch seeds without metadata exactly as before, and the Settings refresh backfills).
- **Per-model pricing rides to the worker as `R_TASK_MODEL_COSTS`**, mirroring the existing `R_TASK_MODEL_CONTEXT_WINDOWS` plumbing, and lands as `cost: {input, output}` (USD per 1M tokens, the models.dev convention) on custom-provider model entries in the generated OpenCode config. Models with unknown pricing get no cost block rather than a fabricated zero. This fixes cost reporting for any custom OpenAI-compatible provider with known pricing, not just the trial.

## Tests

- types: cost emission on model entries + cost-map parser edge cases
- db: resolver emits `R_TASK_MODEL_COSTS` (and context windows) from catalog metadata in gateway mode
- web: roomote→openrouter metadata alias; trial seeding attaches catalog metadata
- worker: env parsed, stripped, and written into the generated config

Note for verification on the existing staging trial tenant: its models were seeded before this fix, so hit **Refresh metadata** in Settings > Models (or re-choose on a fresh workspace) after the rollout to pick up pricing; new events then record real cost.

### CODE REVIEW

File: apps/worker/src/run-task/agent-home.ts [hunk 1]
```diff
@@ -1626,6 +1632,7 @@ function resolveModelBackedOpenCodeConfig(
                 openAiCompatibleModelIds,
                 visionModel ?? effectiveCodingModel,
                 modelContextWindows,
+                modelCosts,
```
[the remark this answers is not in the document]
Comment (mrubens) [reply apps/worker/src/run-task/agent-home.ts:1638]: Fixed: openAiCompatibleModelIds now also includes Object.keys(modelCosts), with a test covering a price-only switchable model receiving its cost block.
